tests/integration/test_submit.py

```diff
@@ -0,0 +1,111 @@
+import json
+import os
+import re
+from pathlib import Path
+from traceback import print_exception
+
+import pytest
+from click.testing import CliRunner
+
+from cromshell.__main__ import main_entry as cromshell
+
+
+class TestSubmit:
+    workflow_id = None  # Anyway to avoid using global variables?
+
+    @pytest.mark.parametrize(
+        "wdl, json_file, exit_code",
+        [
+            ("tests/workflows/helloWorld.wdl", "tests/workflows/helloWorld.json", 0),
+            ("tests/workflows/helloWorld.wdl", "tests/workflows/not_valid.json", 1),
+            ("tests/workflows/not_valid.wdl", "tests/workflows/helloWorld.json", 1),
+        ],
+    )
+    def test_submit(
+        self, local_cromwell_url: str, wdl: str, json_file: str, exit_code: int
+    ):
+        runner = CliRunner(mix_stderr=False)
+        # The absolute path will be passed to the invoke command because
+        # the test is being run in temp directory created by CliRunner.
+        absolute_wdl = str(Path(wdl).resolve())
+        absolute_json = str(Path(json_file).resolve())
+        with runner.isolated_filesystem():
+            result = runner.invoke(
+                cromshell,
+                [
+                    "--cromwell_url",
+                    local_cromwell_url,
+                    "submit",
+                    absolute_wdl,
+                    absolute_json,
+                ],
+            )
+            assert result.exit_code == exit_code, (
+                f"\nSTDOUT:\n{result.stdout}"
+                f"\nSTDERR:\n{result.stderr}"
+                f"\nExceptions:\n{result.exception}"
+                f"\n{print_exception(*result.exc_info)}"
+            )
+
+            # If submission passed get workflow id and set workflow id global variable
+            if exit_code == 0:
+                # Using regex to match json output from stdout
+                pattern = "{.*}"
+                try:
+                    result_stdout_substring = re.search(
+                        pattern, result.stdout, re.DOTALL  # Dotall to match '\n'
+                    ).group(0)
+                except AttributeError:
+                    print("Trouble extracting json in stdout")
+                    print(repr(result.stdout))
+                    print(result_stdout_substring)
+
+                stdout_substring_formatted = json.loads(result_stdout_substring)
+                TestSubmit.workflow_id = stdout_substring_formatted["id"]
+
+    def test_workflow_id_in_db(
```

Comment:
Good to combine tests here, to avoid global variables with tests that are linked.